Meta/CMake/jakt.cmake

```diff
@@ -76,6 +76,59 @@ if (NOT ENABLE_JAKT)
     return()
 endif()
 
+# Get the C++ compiler's include dirs
+# Equivalent to: $CXX -xc++ /dev/null -E -Wp,-v 2>&1 | sed -n 's,^ ,,p'
+execute_process(
+    COMMAND ${CMAKE_CXX_COMPILER} -xc++ /dev/null -E
+    OUTPUT_VARIABLE CXX_INCLUDES
+    ERROR_VARIABLE CXX_INCLUDES
+)
+string(REGEX MATCHALL "(^|\n) [^\n]*" CXX_INCLUDES "${CXX_INCLUDES}")
+string(REGEX REPLACE "\n" "" CXX_INCLUDES "${CXX_INCLUDES}")
+string(REGEX REPLACE "; +" ";" CXX_INCLUDES "${CXX_INCLUDES}")
+
+# Get the builtin/system includes
+# Equivalent to $CXX -E test.cpp |& sed -e '/^# 1 /p;d' | sed -e 's/# 1 "//;s/"[^"]*$//'
+file(WRITE ${CMAKE_BINARY_DIR}/jakt_test.cpp "#include <new>")
```

Comment:
I mean, this works. But would it be easier to understand if it was a python script instead?

I'm scared of regexes :)

Also why are there two commands? Surely This is good enough?

```
$ ./Toolchain/Local/aarch64/bin/aarch64-pc-serenity-gcc  -xc++ -E -Wp,-v - < /dev/null
ignoring nonexistent directory "/Users/andrew/Source/serenity/Toolchain/../Build/aarch64/Root/usr/local/include"
#include "..." search starts here:
#include <...> search starts here:
 /Users/andrew/Source/serenity/Toolchain/Local/aarch64/lib/gcc/aarch64-pc-serenity/13.2.0/../../../../aarch64-pc-serenity/include/c++/13.2.0
 /Users/andrew/Source/serenity/Toolchain/Local/aarch64/lib/gcc/aarch64-pc-serenity/13.2.0/../../../../aarch64-pc-serenity/include/c++/13.2.0/aarch64-pc-serenity
 /Users/andrew/Source/serenity/Toolchain/Local/aarch64/lib/gcc/aarch64-pc-serenity/13.2.0/../../../../aarch64-pc-serenity/include/c++/13.2.0/backward
 /Users/andrew/Source/serenity/Toolchain/Local/aarch64/lib/gcc/aarch64-pc-serenity/13.2.0/include
 /Users/andrew/Source/serenity/Toolchain/Local/aarch64/lib/gcc/aarch64-pc-serenity/13.2.0/include-fixed
 /Users/andrew/Source/serenity/Toolchain/Local/aarch64/lib/gcc/aarch64-pc-serenity/13.2.0/../../../../aarch64-pc-serenity/include
 /Users/andrew/Source/serenity/Toolchain/../Build/aarch64/Root/usr/include
End of search list.
# 0 "<stdin>"
# 0 "<built-in>"
# 0 "<command-line>"
# 1 "<stdin>"
```

```python
got_dotdotdot = False
got_bracket = False
paths = []
for path.strip() in lines:
    if not got_dotdotdot:
        if path == "#include "..." search starts here:"
            got_dotdotdot = True
         continue
    if path ==  "#include <...> search starts here:"
        got_bracket = True
        continue
    paths.append(pathlib.Path(path.split(" ")[0]).resolve())
``` 
?